src/games/ecards.py

```diff
@@ -0,0 +1,275 @@
+        content_dict = {
+            pid: self.embed(
+                f"This round's prompt: {prompt}\nYou're the leader for this round - sit back and relax!"
+            )
+            if pid == leader
+            else self.embed(f"This round's prompt: {prompt}")
+            for pid in pids
+        }
+
+        # get all message responses
+        tasks = []
+        for pid in pids:
+            # get channel
+            dm_channel = await self.players[pid].create_dm()
+
+            if pid != leader:
+                # make pipeline
+                ipl = InteractionPipeline(ChoiceInteraction(*hands[pid], max_votes=1))
+
+                # send message by storing coroutine
+                tasks.append(ipl.send_and_watch(dm_channel, content_dict[pid]))
+            else:
+                # leader
+                tasks.append(dm_channel.send(embed=content_dict[pid]))
+
+        # await routines
+        dm_response = {
+            # ensure type consistent output
+            pid: resp if pid != leader else {}
+            for pid, resp in zip(pids, await asyncio.gather(*tasks))
+        }
+
+        # unpack which card played
+        # pid -> str
+        cards_played = {} 
+        for pid, resp in dm_response.items():
+
+            # make sure result is present
+            if "choice" in resp.get("response", {}):
+                # invert choices:
+                inverted = {v: k for k, v in resp["response"]["choice"].items()}
+
+                # get, or falsey (min enumeration is 1)
+                index = inverted.get(1, 0)
+
+                if index:
+                    index = inverted[1] - 1
+                    cards_played[pid] = hands[pid].pop(index)
+                    self.logging.info(f"player {pid} chose {index}")
+
+                else:
+                    # no card: log for now
+                    self.logging.info(
+                        f"Player {self.players[pid]} did not play a card."
+                    )
+
+            elif pid != leader:
+                # log warning
+                self.logging.warning(f"Bad response from {self.players[pid]}: {resp}")
+
+        # shuffle responses to list
+        shuffled_responses = list(cards_played.values())
+        random.shuffle(shuffled_responses)
+
+        asyncio.sleep(self.wait_duration)
+
+        if len(shuffled_responses) == 0:
+            # No-one played a card - skip the round
+            await self.channel.send(
+                embed=self.embed(
+                    "No-one played a card. Are the players even there? Skipping this round..."
+                )
+            )
+        elif len(shuffled_responses) == 1:
+            # Only one person played a card - award them the victory
+            winning_card = shuffled_responses[0]
+            win_generator = filter(lambda i: i[1] == winning_card, cards_played.items())
+            if win_generator:
+                # get key
+                winning_pid = next(win_generator)[0]
```

Comment:
This here is repeated over lines +233 to +244 -- seperate into a result extracting method?